### PR TITLE
Created New Task view

### LINKED ISSUE #108: Create new task view
Create new task view

### PR DESCRIPTION
**Pull request by aradz1997**
Created the new task view, when a manager enters the site he
 will see on the navbar the button "new task" and can enter the page.

1) I used the 'created new task' function so all the validations are the same.
2) also used bootstrap to make it look nicer
3) I used 'messages' to notify the manager if the task was added successfully or something went wrong.
4) a cool thing I find is when something is not well with the form it will keep all the details you already added and not erase every thing

fixes #108

### CODE REVIEW

File: tasks/templates/new_task.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{% extends 'base.html' %}
+{% block content %}
+    <div class="container" style="width: 60%">
+        <div class="mb-3">
+            <div class="newfrm">
+                <p class="fs-1">NEW TASK :</p>
+
```
Comment (kobihk) [inline tasks/templates/new_task.html:7]: do we need this empty line?

File: tasks/templates/new_task.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+                {% if messages %}
+
```
Comment (kobihk) [inline tasks/templates/new_task.html:9]: same ^^

File: tasks/templates/new_task.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                    <ul class="messages" style="color: red">
+                        {% for message in messages %}
+                            <li class="{{ message.tags }}">{{ message }}</li>
+                        {% endfor %}
+                    </ul>
+                    <br>
+                {% endif %}
+
+
```
Comment (kobihk) [inline tasks/templates/new_task.html:18]: same ^^

File: tasks/templates/new_task.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                <form method="POST">
+                    {% csrf_token %}
+                    {{ form.as_p }}
+                    <input type="submit" value="Create Task">
+                </form>
+            </div>
+        </div>
+    </div>
+
+{% endblock content %}
```
Comment (kobihk) [inline tasks/templates/new_task.html:28]: please add a new line at the end of the file

File: tasks/views.py [hunk 5]
```diff
@@ -1,3 +1,33 @@
...
 # Create your views here.
+def new_task(request):
+    if request.method == "POST":
+        form = TaskForm(request.POST)
+        if form.is_valid():
+            taskForm = form.save(commit=False)
+            try:
+                Task.create_task(taskForm.title, taskForm.assignee, taskForm.created_by, taskForm.priority,
+                                 taskForm.status,
+                                 taskForm.description)
```
Comment (kobihk) [inline tasks/views.py:17]: I think you can write it in the same line as you did it in the line above.
please be consistent in your code.
```suggestion
                                 taskForm.status, taskForm.description)
```

File: tasks/views.py [hunk 6]
```diff
@@ -1,3 +1,33 @@
...
+                    initial={'title': taskForm.title, 'assignee': taskForm.assignee, 'created_by': taskForm.created_by,
+                             'priority': taskForm.priority,
+                             'status': taskForm.status, 'description': taskForm.description})
+
+    else:
+        form = TaskForm()
+
+    return render(request, 'new_task.html', {
+        'form': form
+    })
```
Comment (kobihk) [inline tasks/views.py:33]: Please do it in one line:
```suggestion
    return render(request, 'new_task.html', {'form': form})
```

**Review by kobihk — CHANGES_REQUESTED** [review]

**Comment by aradz1997** [comment]
commit - "Adding feature to created_by" is for
Adding to the "created_by" field  a restriction that the user can see in the drop-down list only managers

File: users/models.py [hunk 7]
```diff
@@ -51,6 +51,10 @@ def check_valid_name(string):
         if(len(string) > 30 or not string.isalpha()):
             raise ValueError
 
+    @classmethod
+    def get_all_managers(cls):
```
Comment (orzionpour) [inline users/models.py:55]: Please add tests for this function

**Review by orzionpour — CHANGES_REQUESTED** [review]

File: users/models.py (hunk 7 above)
Comment (aradz1997) [reply users/models.py]: deleted this function

**Comment by aradz1997** [comment]
Changed the DDL of the 'Created_by' to be only the manager that is logged in

**Comment by kobihk** [comment]
@aradz1997 please avoid using force-push 
because for the code reviewers it forces them to go all over the code from the beginning instead of seeing only the diff between the latest CR to the resolved code

**Review by kobihk — APPROVED** [review]

**Review by orzionpour — APPROVED** [review]

**Comment by aradz1997** [comment]
> @aradz1997 please avoid using force-push because for the code reviewers it forces them to go all over the code from the beginning instead of seeing only the diff between the latest CR to the resolved code

yes I tried, but after merging PR 110 I had a lot of conflicts and all the commits got messed out, sorry about it

**Review by Asafh7 — APPROVED** [review]

**Pull request merged**